Tests/SonarScanner.MSBuild.Shim.Test/AdditionalFilesServiceTest.cs

```diff
@@ -0,0 +1,88 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using NSubstitute;
+using SonarScanner.MSBuild.Common;
+
+namespace SonarScanner.MSBuild.Shim.Test;
+
+[TestClass]
+public class AdditionalFilesServiceTest
+{
+    private readonly IDirectoryWrapper wrapper;
+    private readonly DirectoryInfo directoryInfo;
+    private readonly AdditionalFilesService sut;
+
+    public AdditionalFilesServiceTest()
+    {
+        wrapper = Substitute.For<IDirectoryWrapper>();
+        sut = new(wrapper);
+        directoryInfo = new DirectoryInfo(Directory.GetCurrentDirectory());
+    }
+
+    [TestMethod]
+    public void AdditionalFiles_NoExtensionsFound_ReturnsEmpty()
+    {
+        var files = sut.AdditionalFiles(new() { ServerSettings = [] }, directoryInfo);
+
+        files.Should().BeEmpty();
+        wrapper.DidNotReceive().EnumerateFiles(Arg.Any<string>(), Arg.Any<string>(), Arg.Any<SearchOption>());
+    }
+
+    [DataTestMethod]
+    [DataRow("sonar.tsql.file.suffixes")]
+    [DataRow("sonar.plsql.file.suffixes")]
+    [DataRow("sonar.yaml.file.suffixes")]
+    [DataRow("sonar.xml.file.suffixes")]
+    [DataRow("sonar.json.file.suffixes")]
+    [DataRow("sonar.css.file.suffixes")]
+    [DataRow("sonar.html.file.suffixes")]
+    [DataRow("sonar.javascript.file.suffixes")]
+    [DataRow("sonar.typescript.file.suffixes")]
+    public void AdditionalFiles_ExtensionsFound_SingleProperty_ReturnsFiles(string propertyName)
```

Comment:
Could you please also add a test with a configured extension that is missing the `.` e.g. "sql" instead of ".sql"

Reply:
I added it on the already existing tests, as I had 2 extensions already in each.